scripts/lib/api/htmlToMd.test.ts

```diff
@@ -987,3 +989,113 @@ By default this is sys.stdout.</p></li>
+      <dt class="sig sig-object py" id="qiskit.test.myClass2">
+      <em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">qiskit.test.</span></span><span class="sig-name descname"><span class="pre">myClass2</span></span><span class="sig-paren">()</span></dt>
+      <dd><p>Inline class 2</p>
+
+        <dl class="py class">
+        <dt class="sig sig-object py" id="qiskit.test.myClass2.myClass3">
+        <em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">qiskit.test.</span></span><span class="sig-name descname"><span class="pre">myClass3</span></span><span class="sig-paren">()</span></dt>
+        <dd><p>Inline class inside another class</p>
+
+        <dl class="py method">
+        <dt class="sig sig-object py" id="qiskit.test.myClass2.myClass3.method3">
+        <em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">method3</span></span><span class="sig-paren">()</span></dt>
+        <dd><p>Method 3</p>
+        </dd></dl>
+
+        <dl class="py method">
+        <dt class="sig sig-object py" id="qiskit.test.myClass2.myClass3.method4">
+        <em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">method4</span></span><span class="sig-paren">()</span></dt>
+        <dd><p>Method 4</p>
+        </dd></dl>
+        
+        </dd></dl>
+      </dd></dl>
+
+      <dl class="py method">
+      <dt class="sig sig-object py" id="qiskit.test.method5">
+      <em class="property"><span class="pre">abstract</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">method5</span></span><span class="sig-paren">()</span></dt>
+      <dd><p>Method outside the classes</p>
+      </dd></dl>
+      </div>
+      `,
+    ),
+  ).toEqual(`# Test classes and methods
+
+### myClass1
+
+<Class id="qiskit.test.myClass1" signature="qiskit.test.myClass1()">
+  Inline class
+
+  #### method1
+
+  <Function id="qiskit.test.myClass1.method1" signature="abstract method1()">
+    Method 1
+  </Function>
+
+  #### method2
+
+  <Function id="qiskit.test.myClass1.method2" signature="abstract method2()">
+    Method 2
+  </Function>
+</Class>
+
+### myClass2
+
+<Class id="qiskit.test.myClass2" signature="qiskit.test.myClass2()">
+  Inline class 2
+
+  #### myClass3
+
+  <Class id="qiskit.test.myClass2.myClass3" signature="qiskit.test.myClass3()">
+    Inline class inside another class
+
+    ##### method3
+
+    <Function id="qiskit.test.myClass2.myClass3.method3" signature="abstract method3()">
+      Method 3
+    </Function>
+
+    ##### method4
+
+    <Function id="qiskit.test.myClass2.myClass3.method4" signature="abstract method4()">
+      Method 4
+    </Function>
+  </Class>
+</Class>
+
+### method5
```

Comment:
This would be a function, not a method

Could you please add an Exception as well? It'd be good for it to be under an h3 header to test that we set it to h4

Reply:
Done! I added two exceptions inside the classes with different ordering to test that we always set the correct heading level. I also added an attribute in between the classes :+1: